core/src/com/unciv/ui/trade/DiplomacyScreen.kt

```diff
@@ -258,6 +267,44 @@ class DiplomacyScreen(val viewingCiv:CivilizationInfo):CameraStageBaseScreen() {
         diplomacyTable.add(backButton)
         return diplomacyTable
     }
+    private fun getImprovementGiftTable(otherCiv: CivilizationInfo): Table{
+        val improvementGiftTable = getCityStateDiplomacyTableHeader(otherCiv)
+        improvementGiftTable.addSeparator()
+
+        val improvableTiles = otherCiv.getCapital().getImprovableTiles().filterNot {it.getTileResource().resourceType == ResourceType.Bonus}.toList()
+        val tileImprovements = otherCiv.gameInfo.ruleSet.tileImprovements.filter { it.value.turnsToBuild != 0 }
+        var anyImprovements = false
+
+        for (improvableTile in improvableTiles){
+            for (tileImprovement in tileImprovements.values){
+                if (improvableTile.canBuildImprovement(tileImprovement, otherCiv) && improvableTile.getTileResource().improvement == tileImprovement.name){
+                    anyImprovements = true
+                    val improveTileButton = "Build $tileImprovement on ${improvableTile.getTileResource()} (200 Gold)".toTextButton()
```

Comment:
This should be with templates - "build [improvement] on [resource] ([statCost])" - for translation

Reply:
did i do the templates correctly? btw the 200 gold is constant